diff --git a/configure.ac b/configure.ac
index 185f175..e43d4f8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -8,7 +8,7 @@ VALA_REQUIRED=0.7.0
 GLIB_REQUIRED=2.12.0
 GTK_REQUIRED=2.14.0
 LIBXML_REQUIRED=2.7.0
-LIBNOTIFY_REQUIRED=0.7.0
+LIBNOTIFY_REQUIRED=0.5.0
 
 AC_PROG_CC
 AM_PROG_CC_C_O
diff --git a/src/configuration_list.vala b/src/configuration_list.vala
index 50fbee7..2c42847 100644
--- a/src/configuration_list.vala
+++ b/src/configuration_list.vala
@@ -89,10 +89,10 @@ namespace VDEPN {
 			// file
 			Menu file_menu = new Menu ();
 			ImageMenuItem file_item = new ImageMenuItem.with_mnemonic ("_File");
-			ImageMenuItem new_conn_item = new ImageMenuItem.from_stock (Gtk.Stock.NEW, accel_group);
-			ImageMenuItem save_conn_item = new ImageMenuItem.from_stock (Gtk.Stock.SAVE, accel_group);
-			ImageMenuItem rm_conn_item = new ImageMenuItem.from_stock (Gtk.Stock.DELETE, accel_group);
-			ImageMenuItem exit_item = new ImageMenuItem.from_stock (Gtk.Stock.QUIT, accel_group);
+			ImageMenuItem new_conn_item = new ImageMenuItem.from_stock (Gtk.STOCK_NEW, accel_group);
+			ImageMenuItem save_conn_item = new ImageMenuItem.from_stock (Gtk.STOCK_SAVE, accel_group);
+			ImageMenuItem rm_conn_item = new ImageMenuItem.from_stock (Gtk.STOCK_DELETE, accel_group);
+			ImageMenuItem exit_item = new ImageMenuItem.from_stock (Gtk.STOCK_QUIT, accel_group);
 
 			new_conn_item.add_accelerator ("activate", accel_group, (uint) 'n', Gdk.ModifierType.CONTROL_MASK, Gtk.AccelFlags.VISIBLE);
 			save_conn_item.add_accelerator ("activate", accel_group, (uint) 's', Gdk.ModifierType.CONTROL_MASK, Gtk.AccelFlags.VISIBLE);
@@ -203,7 +203,7 @@ namespace VDEPN {
 			/* building help menu */
 			Menu help_menu = new Menu ();
 			ImageMenuItem help_item = new ImageMenuItem.with_mnemonic ("_Help");
-			ImageMenuItem about_item = new ImageMenuItem.from_stock (Gtk.Stock.ABOUT, accel_group);
+			ImageMenuItem about_item = new ImageMenuItem.from_stock (Gtk.STOCK_ABOUT, accel_group);
 			help_menu.append (about_item);
 
 			about_item.add_accelerator ("activate", accel_group, (uint) 'h', Gdk.ModifierType.CONTROL_MASK, Gtk.AccelFlags.VISIBLE);
diff --git a/src/configuration_page.vala b/src/configuration_page.vala
index 7734a6d..d3e0664 100644
--- a/src/configuration_page.vala
+++ b/src/configuration_page.vala
@@ -151,8 +151,8 @@ namespace VDEPN {
 
 			conn_spinner = new Spinner ();
 
-			checkbuttons_box.pack_start (button_ssh);
-			checkbuttons_box.pack_start (button_checkhost);
+			checkbuttons_box.pack_start (button_ssh, true, true, 0);
+			checkbuttons_box.pack_start (button_checkhost, true, true, 0);
 
 			button_ssh.active = config.use_keys;
 			button_checkhost.active = config.checkhost;
@@ -164,9 +164,9 @@ namespace VDEPN {
 			pack_start (user_property, false, false, 0);
 			pack_start (socket_property, false, false, 0);
 			pack_start (ipaddr_property, false, false, 0);
-			pack_start (checkbuttons_box);
-			pack_start (conn_spinner);
-			pack_start (activate_connection);
+			pack_start (checkbuttons_box, true, true, 0);
+			pack_start (conn_spinner, true, true, 0);
+			pack_start (activate_connection, true, true, 0);
 
 			/* tries to activate the connection, showing a fancy
 			 * spinner while the Application works in background */
@@ -223,6 +223,7 @@ namespace VDEPN {
 
 							/* it's enough, I hate spinners. BURN'EM WITH FIRE */
 							conn_spinner.stop ();
+							return null;
 						}, false);
 
 					/* Make the button sensible to signals again */
